pycalphad/mapping/strategy/step_strategy.py

```diff
@@ -0,0 +1,170 @@
+from typing import Union
+import logging
+import itertools
+import copy
+
+import numpy as np
+
+from pycalphad import Database, variables as v
+from pycalphad.core.constants import MIN_PHASE_FRACTION
+
+from pycalphad.mapping.primitives import STATEVARS, ZPFLine, Node, Point, ExitHint, Direction, ZPFState
+import pycalphad.mapping.utils as map_utils
+from pycalphad.mapping.strategy.strategy_base import MapStrategy
+
+_log = logging.getLogger(__name__)
+
+class StepStrategy(MapStrategy):
+    def __init__(self, dbf: Database, components: list[str], phases: list[str], conditions: dict[v.StateVariable, Union[float, tuple[float]]], **kwargs):
+        super().__init__(dbf, components, phases, conditions, **kwargs)
+
+    def initialize(self):
+        """
+        Adds initial starting point in middle of free condition
+        """
+        av = self.axis_vars[0]
+        mid_val = (self.axis_lims[av][0] + self.axis_lims[av][1]) / 2
+        mid_conds = copy.deepcopy(self.conditions)
```

Comment:
Need a stepping example. I'm sort of indifferent of whether we use the class API or `stepplot`, but depends on the other comment on whether we keep `stepplot`